### PR TITLE
Remove liquidity pool

### PR DESCRIPTION
**Pull request by AStox**
Makes the `liquidityPoolFactory` fileable on poolManager and adds a `removeLiquidityPool` function

### CODE REVIEW

File: src/PoolManager.sol [hunk 1]
```diff
@@ -430,6 +432,23 @@ contract PoolManager is Auth {
         return liquidityPool;
     }
 
+    function removeLiquidityPool(uint64 poolId, bytes16 trancheId, address currency) public auth {
+        require(pools[poolId].createdAt != 0, "PoolManager/pool-does-not-exist");
+        Tranche storage tranche = pools[poolId].tranches[trancheId];
+        require(tranche.token != address(0), "PoolManager/tranche-does-not-exist");
+        require(isAllowedAsInvestmentCurrency(poolId, currency), "PoolManager/currency-not-supported");
```
Comment (hieronx) [inline src/PoolManager.sol:439]: I don't think this check is needed
```suggestion
```

File: src/PoolManager.sol [hunk 2]
```diff
@@ -430,6 +432,23 @@ contract PoolManager is Auth {
...
+        address liquidityPool = tranche.liquidityPools[currency];
+        require(liquidityPool != address(0), "PoolManager/liquidity-pool-not-deployed");
+
+        delete tranche.liquidityPools[currency];
+
+        AuthLike(address(investmentManager)).deny(liquidityPool);
+
+        AuthLike(tranche.token).deny(liquidityPool);
+        TrancheTokenLike(tranche.token).removeTrustedForwarder(liquidityPool);
+    }
```
Comment (hieronx) [inline src/PoolManager.sol:450]:
```suggestion
    escrow.approve(tranche.token, liquidityPool, 0);

    }
```

File: test/PoolManager.t.sol [hunk 3]
```diff
@@ -485,6 +489,117 @@ contract PoolManagerTest is TestSetup {
...
+    function testRemoveLiquidityPool()public {
+        address lPool_ = deploySimplePool();
+        LiquidityPool lPool =  LiquidityPool(lPool_);
+        uint64 poolId = lPool.poolId();
+        bytes16 trancheId = lPool.trancheId();
+        address currency = address(lPool.asset());
+
+        root.relyContract(address(poolManager), address(this));
+        poolManager.removeLiquidityPool(poolId, trancheId, currency);
+        assertEq(poolManager.getLiquidityPool(poolId, trancheId, currency), address(0));
```
Comment (hieronx) [inline test/PoolManager.t.sol:501]: Would probably be good to add some checks here that after removing, the old LP is not a ward anymore on the investment mgr or tranche token, nor a trusted forwarder, and has 0 approval on the tranche token

File: test/PoolManager.t.sol [hunk 4]
```diff
@@ -485,6 +489,117 @@ contract PoolManagerTest is TestSetup {
         centrifugeChain.updateTrancheTokenPrice(poolId, trancheId, currencyId, price, uint64(block.timestamp - 1));
     }
 
+    function testRemoveLiquidityPool()public {
```
Comment (hieronx) [inline test/PoolManager.t.sol:492]:
```suggestion
    function testRemoveLiquidityPool() public {
```

File: src/PoolManager.sol [hunk 5]
```diff
@@ -430,6 +432,23 @@ contract PoolManager is Auth {
...
+        address liquidityPool = tranche.liquidityPools[currency];
+        require(liquidityPool != address(0), "PoolManager/liquidity-pool-not-deployed");
+
+        delete tranche.liquidityPools[currency];
+        escrow.approve(address(tranche.token), liquidityPool, 0);
+
+        AuthLike(address(investmentManager)).deny(liquidityPool);
+
+        AuthLike(tranche.token).deny(liquidityPool);
+        TrancheTokenLike(tranche.token).removeTrustedForwarder(liquidityPool);
```
Comment (hieronx) [inline src/PoolManager.sol:449]: We should emit an event here
```suggestion
        TrancheTokenLike(tranche.token).removeTrustedForwarder(liquidityPool);
        
        emit RemoveLiquidityPool(poolId, trancheId, currency, liquidityPool);```

File: test/PoolManager.t.sol (hunk 4 above)
Comment (AStox) [reply test/PoolManager.t.sol:492]: surprised `forge fmt` didn't catch this one

File: src/PoolManager.sol [hunk 6]
```diff
@@ -430,6 +435,25 @@ contract PoolManager is Auth {
...
```
Comment (hieronx) [inline src/PoolManager.sol:452]: To match the order in `deployLiquidityPool` so it's easier to compare
```suggestion

        delete tranche.liquidityPools[currency];
        
        AuthLike(address(investmentManager)).deny(liquidityPool);

        AuthLike(tranche.token).deny(liquidityPool);
        TrancheTokenLike(tranche.token).removeTrustedForwarder(liquidityPool);
        
        escrow.approve(address(tranche.token), liquidityPool, 0);
```

**Review by hieronx — APPROVED** [review]
🚀

**Review by ilinzweilin — APPROVED** [review]

**Review by hieronx — APPROVED** [review]

**Pull request merged**